website/scripts/ros/ros_traffic_signal.js

```diff
@@ -1,67 +1,75 @@
-            /**
-             *  UNLIT=0; GREEN=1; YELLOW=2; RED=3; FLASHING_GREEN=4; FLASHING_YELLOW=5; FLASHING_RED=6
-            */
-           //Prevent repeating the same state
-           if(signalState != element.state)
-           {
-                signalState = element.state;
-                switch(signalState)
-                {
-                    case SIGNAL_GREEN_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('green',element.remaining_time));
-                        break;
-                    case SIGNAL_RED_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('red',element.remaining_time));
-                        break;
-                    case SIGNAL_YELLOW_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('yellow',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_GREEN_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_green',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_YELLOW_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_yellow',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_RED_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_red',element.remaining_time));
-                        break;
-                    default:
-                        $('.traffic-signal-col').html('');
-                        $('.traffic-signal-col').append(updateTrafficSignal('',''));
-                        console.error("Traffic signal state is invalid");
-                        break;
-                } 
-                //set back to black after 5 seconds.
-                if(!isTimeerSet)
-                {
-                    isTimeerSet=true;
-                    setTimeout(function(){
-                        $('.traffic-signal-col').html('');
-                        $('.traffic-signal-col').append(updateTrafficSignal('',''));
-                        isTimeerSet=false;
-                        signalState = null;
-                    }, 5000);
-                }
-           }      
-        });
-            
+         try{
+            if(Array.isArray(message.intersections.intersection_state_list)){
+                message.intersections.intersection_state_list.forEach(element=>{
+                    element.states.movement_list.forEach(inner_ele=>{
+                        inner_ele.state_time_speed.movement_event_list.forEach(event_ele=>{
+                            let signal_state = event_ele.event_state.movement_phase_state;
+                            let remaining_time = event_ele.timing.min_end_time;
+                            //Prevent repeating the same state
+                            if(signalState !=signal_state)
```

Comment:
The way this is written the signalState will be randomly set based on the order of lights in the intersection. I don't see how we would know this is showing the correct light or even intersection. It seems like the intersection/signal group would need to be specified or all the lights would need to be shown.

Reply:
I am assuming the spat is published by on intersection. If there are multiple elements in the movement, or movement phase lists, it will always replaced by the last one.  It only shows one light at a time.

Reply:
How do we know which intersection is of interest? I mean is there a indicator in the spat message that says this is the intersection that the vehicle is approaching?

Reply:
I am not sure what role does this signalgroup play here.